### PR TITLE
analyze: add test for verify exit status with warnings

### PR DESCRIPTION
**Pull request by ryantimwilson**
This adds an integration test for `systemd-analyze verify` to prevent regressions like https://github.com/systemd/systemd/pull/22989

Specifically, we add a test to verify a unit file with only warnings will return exit code 0 if --recursive-errors is not specified and nonzero if --recursive-errors=yes/no/one

### CODE REVIEW

File: test/units/TEST-65-ANALYZE.sh [hunk 1]
```diff
@@ -381,6 +381,29 @@ systemd-analyze verify /tmp/multi-exec-start.service
 echo 'ExecStart=command-should-not-exist' >>/tmp/multi-exec-start.service
 (! systemd-analyze verify /tmp/multi-exec-start.service)
 
+# Prevent regression from #20233 where systemd-analyze will return nonzero exit codes on warnings
+
+# Unit file with warning "Unknown key name 'foo' in section 'Unit', ignoring"
+cat <<EOF >/tmp/testwarnings.service
+[Unit]
+foo = bar
```
Comment (daandemeyer) [inline test/units/TEST-65-ANALYZE.sh]: We don't tend to put whitespace between unit assigments so please drop those. in other words make this:

```conf
[Unit]
Foo=Bar

[Service]
ExecStart=echo hello
```

**Review by DaanDeMeyer — APPROVED** [review]

File: test/units/TEST-65-ANALYZE.sh (hunk 1 above)
Comment (ryantimwilson) [reply test/units/TEST-65-ANALYZE.sh]: Fixed in latest version!

**Comment by ryantimwilson** [comment]
@DaanDeMeyer addressed comment in most recent version and re-requested review

Some tests are failing for unrelated reasons (e.g. python packaging issue we discussed earlier), should I rerun them?

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
No need to rerun, Arch is being fixed in another PR and CentOS should fix itself hopefully today or tomorrow

**Pull request merged**